nanopub/nanopub.py

```diff
@@ -200,114 +207,111 @@ class NanopubClient:
-            searchparams['obj'] = obj
+            params['obj'] = obj
 
-        return self._search(searchparams=searchparams,
-                            max_num_results=max_num_results, apiurl=apiurl)
+        return self._search(endpoint='find_nanopubs_with_pattern',
+                            params=params,
+                            max_num_results=max_num_results)
 
-    def search_things(self, thing_type=None, searchterm=' ',
-                      max_num_results=1000, apiurl='http://grlc.nanopubs.lod.labs.vu.nl/api/local/local/find_things'):
+    def find_things(self, type=None, searchterm=' ',
+                    max_num_results=1000):
         """
         Searches the nanopub servers (at the specified grlc API) for any nanopubs of the given type, with given search term,
         up to max_num_results.
         """
+        if not type or not searchterm:
+            raise ValueError(f'type and searchterm must BOTH be specified in calls to'
+                             f'Nanopub.search_things. type: {type}, searchterm: {searchterm}')
 
-        searchparams = {}
-        if not thing_type or not searchterm:
-            print(f"Received thing_type='{thing_type}', searchterm='{searchterm}'")
-            raise ValueError('thing_type and searchterm must BOTH be specified in calls to Nanopub.search_things')
+        params = dict()
+        params['type'] = type
+        params['searchterm'] = searchterm
 
-        searchparams['type'] = thing_type
-        searchparams['searchterm'] = searchterm
+        return self._search(endpoint='find_things',
+                            params=params,
+                            max_num_results=max_num_results, )
 
-        return self._search(searchparams=searchparams,
-                            max_num_results=max_num_results, apiurl=apiurl)
+    def _query_grlc(self, params, endpoint):
+        """Query the nanopub server grlc endpoint.
 
-    @staticmethod
-    def _search(searchparams=None, max_num_results=None, apiurl=None):
+        Query a nanopub grlc server endpoint (for example: find_text). Try several of the nanopub
+        garlic servers.
         """
-        General nanopub server search method. User should use e.g. search_text() or search_pattern() instead.
-        """
-
-        if apiurl is None:
-            raise ValueError('kwarg "apiurl" must be specified. Consider using search_text() function instead.')
-
-        if max_num_results is None:
-            raise ValueError('kwarg "max_num_results" must be specified. Consider using search_text() function instead.')
-
-        if searchparams is None:
-            raise ValueError('kwarg "searchparams" must be specified. Consider using search_text() function instead.')
-
-
-        # Query the nanopub server for the specified text
         headers = {"Accept": "application/json"}
-        r = requests.get(apiurl, params=searchparams, headers=headers)
+        r = None
+        for grlc_url in self.grlc_urls:
```

Comment:
Do we always try the first server in the list, or should it start from one at random? Perhaps always following the order in the list is better behaviour, even though it will mean one server always being loaded more than the others...?

Reply:
Yes, let's assume we get huge traffic through our client (aim for the moon 😛 ). I will change it to pick a random one.